### ISSUE TITLE
Deprecation: set dates for requiring configuration file and notify users

### ISSUE THREAD

**Issue by agjohnson** [readthedocs/readthedocs.org#10351]
This is related to #10348, but separate because this is going to be a larger marketing push instead of a technical change.

With on site notifications enabled, the next step to educating users will be to send some direct emails, write a blog post or two, and probably try any other avenue that we can to reach our users. We talked about reviving some of the work on the RTD assistant code as well, but at very least we should probably expect to jump in and assist some of our larger and older projects.

This is a list of a few of the things that we talked about doing here:

- Decide on some dates.
  - We want a brown out date, with a ~2 hour window where we will fail builds
    - We probably want a feature flag so we can manage which projects get this logic and have a way to disable the logic without a deploy
  - We want a cutoff date, where we will start failing builds for good
    - This likely aligns/precedes the Sept urllib3 deprecation of libssl 1.1.1
- Set some milestones we'll aim for. Re-evaluate our plan if we're wrong
  - Look at adoption numbers, number of projects without a configuraton file
  - Having a timeseries question on metabase would be a great addition
  - We could bump out dates or patch the build images with libssl if our adoption is lagging or we're behind schedule
- Write an email notification to be sent to project maintainers every week or so
  - Include dates, example configuration or link to examples, etc
- Blog post, probably a copy of the email notification in some fashion 
- Mention in newletters too

**Comment by humitos** [comment]
> We probably want a feature flag so we can manage which projects get this logic and have a way to disable the logic without a deploy

I implemented this in https://github.com/readthedocs/readthedocs.org/pull/10355

> Write an email notification to be sent to project maintainers every week or so

I've already implemented this in #10354. We can update the email with more information once we have a blog post and more documentation written around this issue.

**Comment by humitos** [comment]
> Having a timeseries question on metabase would be a great addition

I created the followings, which will be useful to track adoption here:

* [Number of rojects specifying `version: 1` in the config file with a build in the last 30 days](https://ethicalads.metabaseapp.com/question/230-projects-specifying-version-1-config-file?days=30)
* [Timeseries of projects specifying `version: 1` in the config file with a build in the last 90 days](https://ethicalads.metabaseapp.com/question/371-projects-without-readthedocs-yaml-config-file-timeserie?days=90) (takes some time to compute)
* [Number of projects without a config file with a build in the last 30 days](https://ethicalads.metabaseapp.com/question/205-projects-without-readthedocs-yaml-config-file?days=30)

**Comment by agjohnson** [comment]
> [Projects without readthedocs.yaml config file (timeserie)](https://ethicalads.metabaseapp.com/question/371-projects-without-readthedocs-yaml-config-file-timeserie?days=90)

This one seems like the most useful graph for monitoring active builds/projects. It's probably the first graph we'll want to check :+1: 

The version 1 configuration graph is handy, I'm glad that number is fairly low. The non timeseries project count seems like the query is similar to the timeseries graph above, so I'd probably just use that.

I feel like I wanted a different graph here for counting the total number of projects (not builds or projects that are active), but I can't say how that would be used much different than this graph above. Perhaps removing the timeframe from the query? It's okay if this is just a static number that we can note and check up on, it doesn't need to be timeseries.

**Comment by humitos** [comment]
> I feel like I wanted a different graph here for counting the total number of projects (not builds or projects that are active)

We don't have this data outside the `Build` or `BuildData` objects. We need a build (even if it's old) to know if they are building with/without a configuration file. We could check all the projects that have had at least one build since we started collecting data on Telemetry. This may be slow, but I think it's possible. I'll give it a try. I'm also interested in this number 👍🏼

**Comment by agjohnson** [comment]
Yeah, I think that's what I was looking for, though it also seems like that will be close to the timeseries figure too.

**Comment by humitos** [comment]
I created https://ethicalads.metabaseapp.com/question/372-projects-without-readthedocs-yaml-config-file?days=180 which gives us the number of projects that are not using a configuration file. It's based on `BuildData` objects, so it only considers the last 180 days -- which I think it's enough for our purpose [^1]. Currently, this number is **22.9k**

[^1] note that we don't have other way to check whether or not they are using a configuration file than looking into `BuildData` unfortunately.

**Comment by agjohnson** [comment]
Excellent, that seems good enough. We only really need this for a second spot check on overall progress anyways.

**Comment by humitos** [comment]
Once we merge #10354, we will have the **number of total projects without a configuration file** logged in New Relic because of this line: https://github.com/readthedocs/readthedocs.org/blob/400214f7468bb6f4fdf3e7d8171ecf08fc255731/readthedocs/projects/tasks/utils.py#L231-L235

**Comment by agjohnson** [comment]
I've edited with the dates we discussed in our meeting today:

- July 24 first brownout date, times TBD
- Aug 7 second brownout date, times TBD
- Aug 28 is where we may decide to extend our timeframe and patch build images
- Sep 4 is the cutoff date where we start failing builds

**Comment by ericholscher** [comment]
Updated the blog post with those dates: https://github.com/readthedocs/blog/pull/221/

**Comment by humitos** [comment]
I'm going to close this issue because all the work is done here or was split into other issues. We just need to come back with some adoption numbers close to the dates to know how things are going on.

**Comment by agjohnson** [comment]
The date that is next up is our final cut off. Are we comfortable continuing with our plan or do we need to extend this timeframe?

In [May 22.9k projects weren't using a configuration file](https://github.com/readthedocs/readthedocs.org/issues/10351#issuecomment-1568283050), but I don't know where the figure is today as the Metabase question won't load.

@humitos do you have an up to date figure here?

**Comment by humitos** [comment]
IMO, we are fine sticking with our plan. We have been notifying users every week since we started and doing some support to help them migrate to the YAML file.

> In [May 22.9k projects weren't using a configuration file](https://github.com/readthedocs/readthedocs.org/issues/10351#issuecomment-1568283050), but I don't know where the figure is today as the Metabase question won't load.

With the increase of spam we have had it's hard to know how many of those are spam with a simple query 😞 

There are other queries that could help here as well (using 30 days):

* https://ethicalads.metabaseapp.com/question/371-projects-without-readthedocs-yaml-config-file-timeserie?days=30
* https://ethicalads.metabaseapp.com/question/205-projects-without-readthedocs-yaml-config-file?days=30
* https://ethicalads.metabaseapp.com/question/473-projects-not-specifying-build-image-nor-build-os-config-file?days=30
* https://ethicalads.metabaseapp.com/question/372-projects-without-readthedocs-yaml-config-file?days=30
